pkg/imrpc/instance.proto

```diff
@@ -0,0 +1,94 @@
+syntax="proto3";
+
+package imrpc;
+
+import "google/protobuf/empty.proto";
+import "imrpc.proto";
+
+service InstanceService {
+    rpc InstanceCreate(InstanceCreateRequest) returns (InstanceResponse) {}
+	rpc InstanceDelete(InstanceDeleteRequest) returns (InstanceResponse) {}
+	rpc InstanceGet(InstanceGetRequest) returns (InstanceResponse) {}
+	rpc InstanceList(google.protobuf.Empty) returns (InstanceListResponse) {}
+	rpc InstanceLog(InstanceLogRequest) returns (stream LogResponse) {}
+	rpc InstanceWatch(google.protobuf.Empty) returns (stream InstanceResponse) {}
+	rpc InstanceReplace(InstanceReplaceRequest) returns (InstanceResponse) {}
+
+	rpc VersionGet(google.protobuf.Empty) returns(VersionResponse);
+}
+
+message ProcessSpecific {
+	string binary = 1;
+	repeated string args = 2;
+}
+
+message SpdkSpecific {
+	string address = 1;
+    map<string, string> replica_address_map = 2;
+    string frontend = 3;
+	string disk_uuid = 4;
+}
+
+message InstanceSpec {
+	string name = 1;
+	string type = 2;
+	string backend_store_driver = 3;
+	string volume_name = 4;
+	uint64 size = 5;
+
+	int32 port_count = 6;
+	repeated string port_args = 7;
+
+	ProcessSpecific process_specific = 8;
+	SpdkSpecific spdk_specific = 9;
+}
+
+message InstanceStatus {
+	string state = 1;
+	string error_msg = 2;
+	int32 port_start = 3;
+	int32 port_end = 4;
+}
+
+message InstanceCreateRequest {
+	InstanceSpec spec = 1;
+	bool expose_required = 2;
+}
+
+message InstanceDeleteRequest {
+	string name = 1;
+	string type = 2;
+	string backend_store_driver = 3;
+
+	string disk_uuid = 4;
```

Comment:
Does the caller of this API need to provide this? The instance name should be unique and the gRPC server can figure out the disk by itself.

Reply:
The name of a lvols is `diskUUID/replica-name`.
If the diskUUID is not provided, will the SPDK service need to iterate the lvols?